Dynamically determine ETS project number

djexp opened the pull request:
Here is my first pull request on this project, so sorry if I've not followed the expected procedures correctly...

FYI, The change uses regex whilst scanning the ETS project zip file for the prescribed named file as per the KNX spec. 

I think it would be sensible to create a dummy test project file to add to the repo for testing the parser. I'm happy to do this unless there are any strong views on not committing those types of files.

djexp committed:
Dynamically determine ETS project number

sruehl commented:
Thanks for your contribution! :)

chrisdutz commented:
Regarding a knxproj file being added to the project. I would probably suggest not to do that. Even if it's theroretically not a problem. We would be adding a sort-of-binary file to the repo, which might be a problem. But the most important thing is, that it contains the xml descriptors of KNX products. While the project file itself is probably not a problem, these static XML files are probably not licensed under an Apache compatible licence. We would probably get problems with releases if we added such a file. 

If we don't need the static product-descriptors and can live with the project-xml, I would probably opt for manipulating a knxproj file to remove all the stuff that's not a project-xml. In that case I think there probably shouldn't be any issues.

djexp commented:
Hi Chris,

Totally understand the reasons regarding the ets project file not being
added to the repo, kind of limits testing of the parser though. How would
you feel about just adding some 'hand crafted' XML files, just to exercise
the parser?

I've a couple more questions to ask. I'm still experiencing the invalid
decoding of the 2-byte floating point values.. My recollection was you were
going to contact KNX and see if they were aware of this issue, and then I
think we were going to try and progress things with the code generators
together. Was that right?

And secondly, is there a method used elsewhere in PLC4X where it is
possible to specify the local IP to bind to when connecting out? i.e. in
the Knx driver there is no way to specify which network interface to use if
you have multiple on your computer, if there is a prefered way I'm happy to
implement that in the Knx driver. Perhaps this is only relevant to UDP
connections? It's been a fair few years since I've had to write low level
TCP and UDP code.

Many thanks

Darren

On Tue, Feb 22, 2022 at 10:14 AM Christofer Dutz ***@***.***>
wrote:

> Regarding a knxproj file being added to the project. I would probably
> suggest not to do that. Even if it's theroretically not a problem. We would
> be adding a sort-of-binary file to the repo, which might be a problem. But
> the most important thing is, that it contains the xml descriptors of KNX
> products. While the project file itself is probably not a problem, these
> static XML files are probably not licensed under an Apache compatible
> licence. We would probably get problems with releases if we added such a
> file.
>
> If we don't need the static product-descriptors and can live with the
> project-xml, I would probably opt for manipulating a knxproj file to remove
> all the stuff that's not a project-xml. In that case I think there probably
> shouldn't be any issues.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/plc4x/pull/321#issuecomment-1047635314>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AONMLH5Q66AXNZA3DKC72K3U4NOYXANCNFSM5O777GCA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
*Darren Everley - Director*

*Email: ***@***.*** *| Mobile:* 07891405262 *| Website:*
 www.xeropoint.co.uk
*Company: *Xeropoint Ltd. Registered in England and Wales. 11101907
*Address:* Henleaze House, Harbury Road, Bristol, BS9 4PN
